e2ev2/src/utils.js

```diff
@@ -0,0 +1,109 @@
+const assert = require('assert').strict
+const fs = require('fs')
+const https = require('https')
+
+const export_ = o => Object.keys(o).forEach(k => { module.exports[k] = o[k] })
+
+const mapValues =
+  f => obj => Object.keys(obj).reduce((acc, k) => ({...acc, [k]: f(obj[k])}), {})
+export_({mapValues})
+
+const toPairs = xs => xs.reduce((acc, x) => {
```

Comment:
I find a good way to evaluate utilities is to use Node's REPL:
```
% node
Welcome to Node.js v16.15.0.
Type ".help" for more information.
> u = require('./src/utils.js')
{
  mapValues: [Function: mapValues],
  toPairs: [Function: toPairs],
  range: [GeneratorFunction: range],
  id: [Function: id],
  noop: [Function: noop],
  Right: [Function: Right],
  Left: [Function: Left],
  Either: [Function: Either] { fromNullable: [Function (anonymous)] },
  Task: [Function: Task] {
    of: [Function (anonymous)],
    rejected: [Function (anonymous)],
    fromEither: [Function (anonymous)]
  },
  delay: [Function: delay],
  traverse: [Function: traverse],
  promiseEvent: [Function: promiseEvent],
  readFile: [Function: readFile],
  parseJson: [Function: parseJson],
  httpReq: [Function: httpReq],
  readStream: [Function: readStream],
  denseDateTime: [Function: denseDateTime]
}
> u.toPairs([1,2,3])
[ [ 1, 2 ], [ 3 ] ]
> 
```
So your understanding is exactly correct. Indeed this particular function is not used and it may make sense to delete it for now.

Reply:
Thanks for the pointer on Node's REPL. I suspect that that will come in handy.